lib/Frontend/ParseableInterfaceSupport.cpp

```diff
@@ -359,9 +360,17 @@ static bool buildSwiftModuleFromSwiftInterface(
     }
     SerializationOpts.Dependencies = Deps;
     SILMod->setSerializeSILAction([&]() {
-        serialize(Mod, SerializationOpts, SILMod.get());
-      });
+      serialize(Mod, SerializationOpts, SILMod.get());
+    });
+
+    LLVM_DEBUG(llvm::dbgs() << "Running SIL processing passes\n");
+    if (SubInstance.performSILProcessing(SILMod.get())) {
+      LLVM_DEBUG(llvm::dbgs() << "encountered errors\n");
+      SubError = true;
+      return;
+    }
     SILMod->serialize();
```

Comment:
I don't think we need this anymore, since the optimization passes will do it.

(Also we still need to do the thing where we stop running the pipeline after the serialization pass, because we're going to throw it away anyway.)

Reply:
Turns out this was hiding the fact that optimizations weren't being performed at all, because CompilerInvocation::parseArgs was overriding the optimization mode. 🤦‍♂️